crates/optimism/primitives/src/transaction/tx_type.rs

```diff
@@ -257,13 +239,12 @@ mod tests {
         let op_tx = OpTxType(AlloyOpTxType::Legacy);
         let mut buf = BytesMut::new();
         op_tx.encode(&mut buf);
-        assert_eq!(buf, BytesMut::from(&[0x80][..]));
+        assert_eq!(buf, BytesMut::from(&[0xc1, 0x80][..]));
```

Comment:
this now has the rlp header here due to the regular derive trait

Reply:
Yes, @emhane suggested using `RlpDecodable` and `RlpEncodable` derive macros: https://github.com/paradigmxyz/reth/pull/12443#discussion_r1836534465

Is `rlp` header not wanted? Or what is the suggested way?